File: Past projects/Technology of programming 11th grade/Homework/h6/AirPlane.java
// Училище: ТУЕС ( http://www.elsys-bg.org/ )
// Клас: 11 Б
// Номер: 16
// Име:  Едуард Федев Калфов

/*Задача: "Да се създаде програма, която да запълва празните места в самолет,
        като х да е в интервала 1<=х<=3 (пътниците). Всяка група пътници не треябва да са
        разделени от пътеката, или реда. Пример : 100 000 (да), 122 000 (да) ,133 300 (не), 223 330 (не), 221 333 (да)." */

import java.util.Random;

public class AirPlane{
		private int[][] seats = new int[6][27];

		public void add(int rvalue){
			boolean check = false;							//pubic void add(human h)
															//public void remove(human h )
															//public void clear()
															//public int getCapacity()
															//private string gender;
															//public int getMales()
															//public int getFemales()
			for (int j=0;j<27 ;j++ ) {
				for (int i = 0;i < 6 ;i++) {
					if (rvalue==1){
						if (seats[i][j] == 0 ) {
							seats[i][j] = 1;
							check = true;
							break;
						}
					}
					else if (i==6) {
						break;
					}
				
					if (rvalue==2){
						if (i+1==6) {
							continue;
						}

						else if ((seats[i][j] == 0) &&  (seats[i+1][j] == 0) && (i!=2)) {
							seats[i][j] = 2;
							seats[i+1][j] = 2;
							check = true;
							break;
						}
				}
					
				

				if (rvalue==3){
					if (i+2>=6) {
						continue;
					}
					else if ((seats[i][j] == 0) &&  (seats[i+1][j] == 0) && (seats[i+2][j] == 0) && (i!=1) && (i!=2) ) {
						seats[i][j] = 3;
						seats[i+1][j] = 3;
						seats[i+2][j] = 3;
						check = true;
						break;
					}
				}
									

			}
			if(check == true) break;
		}
		}	
		public int free_seats(){
			int count=0;
			for (int j=0;j<27 ;j++ ) {
				for (int i = 0;i < 6 ;i++) {
					if (seats[i][j]==0) {
						count += 1;	
					}

				}
			
			}
			return count;
		}
		public void printSeats(){
	 		for (int i = 0;i < 27 ;i++) {
				for (int j = 0;j < 6 ;j++ ) {
	 	 			System.out.print(seats[j][i]);
	 	 			if(j==2) System.out.print(" ");
	 	 		}
	 	 	System.out.println("");
	 	 	} 
		}
	 public static void main(String[] args) {
	 	
	 	//PRINTING Seats
	 	AirPlane plane = new AirPlane();
	 	plane.printSeats();
	 	 //GENRATING NUMBERS
	 	int total = 0;
	 	int i =0;
	 	System.out.println(plane.free_seats());
		for (;plane.free_seats() > 0 ;i++ ) {
			
			int rvalue = new Random().nextInt(3)+1;
			if(plane.free_seats()<=162){
				plane.add(rvalue);
				
			}
		}
			
			System.out.println("######");
			plane.printSeats();	
			System.out.println(plane.free_seats());
			
			
		
			
			
	}




}

File: Past projects/Technology of programming 11th grade/Homework/h6/Airplane.rb
# Училище: ТУЕС ( http://www.elsys-bg.org/ )
# Клас: 11 Б
# Номер: 16
# Име:  Едуард Федев Калфов

#Задача: "Да се създаде програма, която да запълва празните места в самолет,
#       като х да е в интервала 1<=х<=3 (пътниците). Всяка група пътници не треябва да са
#      разделени от пътеката, или реда. Пример : 100 000 (да), 122 000 (да) ,133 300 (не), 223 330 (не), 221 333 (да)." 

class Airplane
	@@seats = Hash.new {0}
	@@total = 0
	@@check = false

	def	add(rvalue)
		@@check = false
		j = 0
		i = 0
		
		while i<27
			j = 0
			while j<6
				if @@seats[[j,i]] == 0
					if rvalue == 1
						@@seats[[j,i]] = 1
						@@total += rvalue
						@@check = true
						break
					end
					
					if rvalue == 2  && j<=4
						if @@seats[[j+1,i]] == 0 && j%3<=1
							@@seats[[j,i]] = 2
							@@seats[[j+1,i]] = 2
							@@total = @@total + rvalue
							@@check = true
							break
						end
					end
					
					if rvalue == 3 && j<=3
						if @@seats[[j+1,i]] == 0 && @@seats[[j+2,i]] == 0 && j%3 == 0
							@@seats[[j,i]] = 3
							@@seats[[j+1,i]] = 3
							@@seats[[j+2,i]] = 3
							@@total = @@total + rvalue
							@@check = true
							break
						end
					end
					
					j = j + 1
				else
					j = j + 1
				end
			end
			if @@check == true
				i = 27
			else
				i = i + 1
			end
		end
		if @@check == false
			print "The plane is full..."
		end
	end

	def free_seats()
		if @@total < 162
			freeseats = 162 - @@total
			return freeseats
		else 
			puts "There are no freeseats left"
			return 0
		end
	end

	def printSeats()
		for  j in 0..26
				for i in 0..5
					
					print @@seats[[i,j]]
					if i==2
						print " "
						
					end
					
				end
			print "\n"
		end
	end
	
	air = Airplane.new
	people = 0
while air.free_seats() > 0
	people = rand(3) + 1
	air.add(people)
end
	air.printSeats
	puts air.free_seats
end